test/test_optim.py

```diff
@@ -477,6 +477,88 @@ def test_multi_tensor_optimizers(self):
             for p1, p2 in zip(res[0], res[1]):
                 self.assertEqual(p1, p2, atol=5e-5, rtol=0)
 
+    def test_optimizers_foreach_flag(self):
+        if not torch.cuda.is_available():
+            return
+
+        optimizer_pairs_with_flags = [
```

Comment:
This is very nice to have. Maybe we want an OptimizerInfo at some point @jbschlosser haha

Reply:
This function actually just copies `test_multi_tensor_optimizers` and does the same tests when foreach=True, False instead, should I make the below changes to that function as well?